spectator-ext-stackdriver/src/main/java/com/netflix/spectator/stackdriver/StackdriverWriter.java

```diff
@@ -0,0 +1,760 @@
+   */
+  static public TimeSeries
+  findProblematicTimeSeriesElement(String msg, List<TimeSeries> nextN) {
+    String regex = "timeSeries\\[(\\d+?)\\]\\.metric\\.labels\\[(\\d+?)\\]";
+    Matcher matcher = Pattern.compile(regex).matcher(msg);
+    if (matcher.find()) {
+      int tsIndex = Integer.parseInt(matcher.group(1)) - 1;
+      TimeSeries ts = nextN.get(tsIndex);
+        return ts;
+    }
+    return null;
+  }
+
+  /**
+   * Factory method to create a writer.
+   *
+   * The customTypeNamespace is intended to act as a namespace for the
+   * stackdriver metrics found, such as the system name (e.g. spinnaker).
+   * This will help deconflict our metrics with another system that is
+   * also using Stackdriver Custom Metrics in the same project as us and
+   * might want to have CustomTypeDescriptors with the same name.
+   */
+  public static StackdriverWriter makeStackdriverWriter(
+      String customTypeNamespace,
+      String projectName,
+      String applicationName,
+      String applicationVersion,
+      String credentialsPath,
+      MeasurementFilter measurementFilter) {
+    Monitoring monitoring;
+
+    try {
+      HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+      JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+      GoogleCredential credential = loadCredential(transport, jsonFactory,
+                                                   credentialsPath);
+      Monitoring.Builder builder = new Monitoring.Builder(
+            transport, jsonFactory, credential);
+
+        builder.setApplicationName("Spectator");
+      monitoring = builder.build();
+    } catch (IOException|java.security.GeneralSecurityException e) {
+      throw new IllegalStateException(e);
+    }
+
+    log.info("Created StackdriverMonitoring client for project '{}'",
+             projectName);
+    return new StackdriverWriter(monitoring, customTypeNamespace,
+                                 projectName,
+                                 applicationName, applicationVersion,
+                                 measurementFilter);
+  }
+
+  /**
+   * Helper function for the constructor that reads our credentials for
+   * talking to Stackdriver.
+   */
+  private static GoogleCredential loadCredential(
+      HttpTransport transport, JsonFactory factory, String credentialsPath)
+      throws IOException {
+    GoogleCredential credential;
+    if (!credentialsPath.isEmpty()) {
+      FileInputStream stream = new FileInputStream(credentialsPath);
+      credential = GoogleCredential.fromStream(stream, transport, factory)
+                   .createScoped(
+                         Collections.singleton(MonitoringScopes.MONITORING));
+      log.info("Loaded credentials from from {}", credentialsPath);
+    } else {
+      log.info("spectator.stackdriver.monitoring.enabled without"
+               + " spectator.stackdriver.credentialsPath. "
```

Comment:
It looks like the user could have called `makeStackdriverWriter` directly, where does it get tied to the property?